### PR TITLE
NAS-115742 / 22.12 / Add an emby chart

### PR DESCRIPTION
**Pull request by alucryd**
Hi there, I maintain all official emby packages and would like to contribute a chart for scale. What's the process to get into the official scale app repository, is such a PR enough? If so please let me know if you'd like some changes to be made, and if you'd like commits to be squashed. If not please direct me to the official inclusion process documentation.

### CODE REVIEW

**Comment by alucryd** [comment]
Do you guys have access to the full CI error log? Or have any idea why tests fail?

**Review by Qubad786 — COMMENTED** [review]
Thank you @alucryd for your contribution!

Your `helm test` command is failing. It uses `test_values.yaml` to and the `deployment-check.yaml` file to test ( this is how CI runs it https://github.com/truenas/catalog_validation/blob/master/catalog_validation/scripts/charts_validate.py ). If there is any confusion, let me know please and I can assist.

**Comment by Ornias1993** [comment]
> Hi there, I maintain all official emby packages and would like to contribute a chart for scale. What's the process to get into the official scale app repository, is such a PR enough? If so please let me know if you'd like some changes to be made, and if you'd like commits to be squashed. If not please direct me to the official inclusion process documentation.

You're aware there is also already a TrueCharts App available right?
Just to prevent you from spending resources that could be beter spend elsewhere...

**Comment by alucryd** [comment]
> > Hi there, I maintain all official emby packages and would like to contribute a chart for scale. What's the process to get into the official scale app repository, is such a PR enough? If so please let me know if you'd like some changes to be made, and if you'd like commits to be squashed. If not please direct me to the official inclusion process documentation.
> 
> You're aware there is also already a TrueCharts App available right? Just to prevent you from spending resources that could be beter spend elsewhere...

I am, it's not maintained by us, and it's not even using our official docker image, which is why I'm here.

@Qubad786 Thanks for the replies, and apologies for the delay. I'll get back to this next week.

**Comment by Ornias1993** [comment]
Thanks for the info.
The only reason we dont use your official image is that it uses “UID/GID” which is generally not a good practice with kubernetes.

about something not being maintained: so you guys are also committing to maintaining this? (Which is not clear from your PR description, as normaly maintenance falls on IX for IX official Apps).

File: charts/emby/1.0.0/default_values.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Default values
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
```
[the remark this answers is not in the document]
Comment (alucryd) [reply charts/emby/1.0.0/default_values.yaml]: removed

File: charts/emby/1.0.0/templates/tests/deployment-check.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+spec:
+    containers:
+      - name: test-curl
+        image: alpine/curl
+        imagePullPolicy: "IfNotPresent"
+        command:
+          - /bin/sh
+          - -ec
+          - |
+            curl --connect-timeout 5 --max-time 10 --retry 5 --retry-delay 15 --retry-max-time 90 --retry-all-errors -ksf http://{{ $serviceName }}-tcp:8096/emby/System/Ping
```
[the remark this answers is not in the document]
Comment (alucryd) [reply charts/emby/1.0.0/templates/tests/deployment-check.yaml]: Duh, thanks for pointing it out, used 32496 instead

**Comment by alucryd** [comment]
@Ornias1993 Do you have some resource I can read about this UID/GID bad practice? We're using s6 as process manager, and this is what they use, see [0].

We are committing to maintaining it, but we'd have no issue nonetheless with IX shouldering that burden.

@Qubad786 Thanks for all the feedback, finally had some time to address it. Just pushed some changes, if they're not enough I'll spin a minikube locally and try to replicate the validation step.

[0] http://skarnet.org/software/s6/s6-applyuidgid.html

**Comment by Ornias1993** [comment]
> Do you have some resource I can read about this UID/GID bad practice? We're using s6 as process manager, and this is what they use, see [0].

Well, Kubernetes already has native permissions "baked in" and you are forcing users to run their containers as root, only to "down step" to a lower user yourself. Least privilege on Kubernetes is kinda the "best practice".

Basically s6 is designed because docker lacks some features (like init containers with root permissions), but kubernetes already has all those features backed in. It's kindaoff a "Helm Chart light" inside a helm-chart in kubernetes...

**Comment by alucryd** [comment]
@Ornias1993 We need to run as root because this image is not used solely in Kubernetes, and most of our users who just run docker/podman have no clue how to deal with volumes and unix permissions. Having this mechanism lets us automatically chown our data directory, reducing customer support to almost nil regarding permission issues. It downgrades to daemon by default, and more experienced still have the ability to tweak them however they like.

**Comment by Ornias1993** [comment]
> @Ornias1993 We need to run as root because this image is not used solely in Kubernetes, and most of our users who just run docker/podman have no clue how to deal with volumes and unix permissions. Having this mechanism lets us automatically chown our data directory, reducing customer support to almost nil regarding permission issues. It downgrades to daemon by default, and more experienced still have the ability to tweak them however they like.

No idea where this response is coming from  tbh.
You asked me why UID-GID is not a best practice and I just answered your question. Which was in response to me responding on your comment about us not using your official image.

Really don't think you needed to explain s6 overlay to me.
I never said you shouldn't be using it for your image, I said it was not best practice on kubernetes so we aim not to use images that use it.

**Comment by alucryd** [comment]
Not sure where your confusion comes from really. You were the one who explained what s6 does in the first place, I'm saying we need it, and that we're using the env variables s6 itself uses, it's as simple as that. You may not like it and keep using another image and that's fine, but as long as it dosen't break anything it's not realistic for us to change anything. I'm sure you'll understand that we'd rather have our official image in an official package that we may easily change and fix should the need arise.

**Comment by Ornias1993** [comment]
> Not sure where your confusion comes from really. You were the one who explained what s6 does in the first place, I'm saying we need it, and that we're using the env variables s6 itself uses, it's as simple as that. You may not like it and keep using another image and that's fine, but as long as it dosen't break anything it's not realistic for us to change anything. I'm sure you'll understand that we'd rather have our official image in an official package that we may easily change and fix should the need arise.

You asked me.

Also: official means iX Systems Official Apps, not upstream official ;-)

**Review by Qubad786 — APPROVED** [review]
Thank you @alucryd for making the necessary adjustments. LGTM! I'll add an update strategy shortly to it so it can be updated automatically by our bot.

**Comment by alucryd** [comment]
@Qubad786 Thank you very much!

**Pull request merged**